docs/source/making-plugins.rst

```diff
@@ -202,18 +202,94 @@ It is also important to note that source attributes should be set after ``read()
 Driver Discovery
 ----------------
 
+Intake discovers available drivers in three different ways, described below.
+After the discovery phase, Intake will automatically create
+``open_[driver_name]`` convenience functions under the ``intake`` module
+namespace.  Calling a function like ``open_csv()`` is equivalent to
+instantiating the corresponding data-source class.
+
+Entrypoints
+'''''''''''
+
+If you are packaging your driver into an installable package to be shared, you
+should add the following to the package's ``setup.py``:
+
+.. code-block:: python
+
+   setup(
+       ...
+       entry_points={
+           'intake.drivers': [
+               'some_format_name = some_package.and_maybe_a_submodule.YourDriverClass',
```

Comment:
I don't know enough about intake, but it seem that now multiple packages could register the same `some_format_name`, is that of any concern ? 

I also don't remember how entrypoints behave in that case and whether it would only list one entrypoint for that name, of multiples.

Reply:
The `entrypoints` package gives us `entrypoints.get_group_named` (return dict, drop duplicates) and `entrypoints.get_group_all` (return list). In 71c8eb12b4e65c10131046f70133e15c17c3c953 I added code to check for duplicates and debug-log that they are being ignored. 

In 596e96b80cb818927ee52df3331b43089a4b812d I added `autodiscover_all`, which is to `autodiscover` as `get_group_all` is to `get_group_named`. I used it in `intake drivers list` to show all plugins that are discovered but not enabled. They may be not enabled because of a name collision between two or more entrypoints, because of a name collision between entrypoints and package scan results, or because they are explicitly disabled by the configuration file.